### PR TITLE
feat(sim): add X-Plane SITL airframes

### PR DESCRIPTION
**Pull request by alireza787b**
### Summary

Adds community-supported X-Plane SITL support using the external [`px4xplane`](https://github.com/alireza787b/px4xplane) bridge plugin.

This PR includes:
- X-Plane SITL airframe defaults for Cessna 172, TB2, EHang 184, Alia 250, and Quad Tailsitter.
- `make px4_sitl_default xplane_*` launch targets for those airframes.
- PX4 docs for installing the bridge plugin, selecting a target, and configuring the simulator host for local, WSL, Docker, or remote X-Plane setups.
- A community simulator index entry for X-Plane.

### Scope

The PR is intentionally limited to PX4-side airframes, target wiring, and documentation. The bridge plugin, X-Plane aircraft package, and release packaging remain in the external `px4xplane` repository.

### Validation

Local checks run on the cleaned branch:
- `git diff --cached --check` before commit
- Prettier check for the touched docs
- X-Plane airframe metadata generation for the five added airframes
- `make px4_sitl_default`
- Verified the generated Ninja graph contains the five X-Plane targets

### Notes

The branch was rebuilt from current `main` and squashed to one commit to resolve the previous conflicts. Airframe tuning is frozen for this PR unless reviewers request changes.

### CODE REVIEW

File: src/modules/simulation/simulator_mavlink/CMakeLists.txt [hunk 1]
```diff
@@ -71,3 +71,28 @@ add_custom_target(none_iris
 		${PX4_SOURCE_DIR}/ROMFS/px4fmu_common/init.d-posix/airframes/10016_none_iris
 	COMMENT "launching px4 none_iris (SYS_AUTOSTART=10016)"
 )
+
+
+# xplane_c172 (legacy compatibility launch helper)
+add_custom_target(xplane_c172
```
Comment (dagar) [inline src/modules/simulation/simulator_mavlink/CMakeLists.txt:77]: maybe move these to a dedicated file like the others (eg sitl_targets_jsbsim.cmake, etc).

**Comment by ryanjAA** [comment]
Really awesome!

**Comment by ryanjAA** [comment]
are there non-windows plugin compiling instructions? or binaries (for Mac)
thanks! looking forward to trying this

**Comment by alireza787b** [comment]
> are there non-windows plugin compiling instructions? or binaries (for Mac)
> thanks! looking forward to trying this

Well I am working on that ... I will add those instructions soon as well ... I have very limited experience building Cpp in macos...

**Comment by ryanjAA** [comment]
What about in GitHub workflow?

**Comment by alireza787b** [comment]
> What about in GitHub workflow?

I have no experience in that honestly.. but working on that with help of someone...

**Comment by alireza787b** [comment]
Hello again 
I just wanted to have an update on this pull request. should I do anything else before it merges? Since I want to create youtube tutorials on this topic and I prefer not to instruct people use my forked repo. I am waiting until this PR merge then create tutorial based onthis . 
thanks in advance

**Comment by ryanjAA** [comment]
Really looking forward to this! Any update on Mac?

File: src/modules/simulation/simulator_mavlink/CMakeLists.txt (hunk 1 above)
Comment (alireza787b) [reply src/modules/simulation/simulator_mavlink/CMakeLists.txt]: Hello again
I just wanted to have an update on this pull request. should I do anything else before it merges? Since I want to create youtube tutorials on this topic I prefer not to instruct people to use my forked repo. I am waiting until this PR merge then create tutorial based on this.
thanks in advance

**Comment by alireza787b** [comment]
@dagar 

Hello,

I have marked this pull request as ready for review again. It includes the new airframes for the PX4 SITL simulation to interface with X-Plane, specifically the Cessna 172, Ehang 184, and Alia 250 and organized as you requested in last comments.

I am planning to create a YouTube tutorial on this feature, and it would be ideal to have it merged into the official repository to provide clear instructions to users. Could you please review the changes and let me know if any further modifications are required?

Thank you for your support!

Best regards,

File: docs/en/sim_xplane/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+# X-Plane SITL Simulation
+
+X-Plane is a comprehensive flight simulator that can be used with PX4 for Software-In-The-Loop (SITL) simulation. This integration allows testing PX4 with highly realistic flight dynamics across multiple vehicle types.
```
Comment (hamishwillee) [inline docs/en/sim_xplane/README.md]:
```suggestion
:::warning
This simulator is [community supported and maintained](../simulation/community_supported_simulators.md).
It may or may not work with current versions of PX4 and may be removed in future releases.

See [Toolchain Installation](../dev_setup/dev_env.md) for information about the environments and tools supported by the core development team.
:::

X-Plane is a comprehensive flight simulator that can be used with PX4 for Software-In-The-Loop (SITL) simulation.
This integration allows testing PX4 with highly realistic flight dynamics across multiple vehicle types.
```

**Comment by hamishwillee** [comment]
@ryanjAA @dakejahl Who can review this? What are preconditions for this to merge?

Note, I have updated the docs so they will appear in the sidebar and conform to current docs standards, but I'd need another look before we merge.

File: docs/en/sim_xplane/index.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+1. **Frame rate dependency**: X-Plane simulation quality depends on maintaining adequate FPS
+2. **Single aircraft**: One PX4 instance per X-Plane instance
+3. **Sensor simulation**: Limited to basic sensors (no lidar, cameras)
+4. **Operating systems**: px4xplane plugin compatibility varies by platform
+
+## Additional Resources
+
+- **px4xplane Repository**: [https://github.com/alireza787b/px4xplane](https://github.com/alireza787b/px4xplane)
+- **X-Plane Documentation**: [x-plane.com/kb](https://x-plane.com/kb)
+- **PX4 SITL Documentation**: [docs.px4.io/main/en/simulation](https://docs.px4.io/main/en/simulation/)
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]:
```suggestion
- [PX4 SITL Documentation](../simulation/index.md)
```

**Comment by hamishwillee** [comment]
@alireza787b YOu just undid all the changes I made to your docs with the force push.

**Comment by alireza787b** [comment]
@hamishwillee  Sorry about that force push mess. My bad on this one.

  Here's what happened: I saw the "stale PR" notification and thought it got auto-drafted, so I started redesigning the whole procedure from scratch. I've been deep in
   the px4xplane bridge itself for the last 2 weeks - completely redesigned the bridge, workflow, and estimator. Looks like we were both pushing to this branch at the
  exact same time and I didn't notice since I was planning to start fresh anyway. I force updated a bunch of things without checking recent activity first. Lesson learned.
  I've now tried to reorganize everything and add your recommendations back in as much as practical - added the readme, docs, community warning, sidebar integration,
  all the standards you fixed. Should be good now.

  What's left for the future (after this merges):
  - Further PID tuning for the airframes (minor tweaks based on more testing)
  - YouTube tutorial video (want to make it based on official repo, not my custom fork)

  About px4xplane bridge:
  Right now I'm still using a custom bash script with my forked repo. Full instructions are in https://github.com/alireza787b/px4xplane. I'm releasing a new version
  this week - current one is outdated and not working properly. Adding CI/CD and GitHub Actions to it now, should be ready soon. Will get the new release out before
  this PR merges officially.

  Can you review and let me know if I missed anything else or if there's more to fix? Want to make sure this is properly ready.

  Thanks for your patience and for the doc improvements.

File: BRANCH_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
+# PX4 X-Plane SITL Branch Guide
```
Comment (hamishwillee) [inline BRANCH_GUIDE.md]: Branch guide shouldn't be present - at least here - as it would merge into the root but have nothing to do with most of PX4

File: docs/en/sim_xplane/index.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
+# X-Plane SITL Simulation
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]: Please run prettier on all the files you modified under the /docs/en tree.

**Review by hamishwillee — COMMENTED** [review]
Thanks @alireza787b -  I've given it a quick lookover and added a few comments. We need a technical reviewer.

@mahima-yoga Is this something you could look at, or recommend someone else?

**Comment by alireza787b** [comment]
@hamishwillee Thank you for the review!

  Both issues have been addressed:
  - ✅ BRANCH_GUIDE.md removed (force-pushed clean history)
  - ✅ Prettier run on all documentation files

  I've also enhanced the documentation based on field experience and streamlined all airframe parameter files for better maintainability.

  Working on a demo video this weekend to showcase the integration.

  Thanks for getting the technical reviewers involved!

File: docs/en/sim_xplane/index.md (hunk 5 above)
Comment (alireza787b) [reply docs/en/sim_xplane/index.md]: Done!

File: BRANCH_GUIDE.md (hunk 4 above)
Comment (alireza787b) [reply BRANCH_GUIDE.md]: Done! Had missed that...

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+# Pitch Rate
+param set-default FW_PR_P 0.365          # P gain
+param set-default FW_PR_I 0.045          # I gain
+param set-default FW_PR_D 0.355          # D gain (high damping)
+param set-default FW_PR_FF 3.4           # Feedforward
+param set-default FW_P_LIM_MAX 20.0      # Max pitch up
+param set-default FW_P_LIM_MIN -15.0     # Max pitch down
+param set-default FW_P_TC 0.45
+param set-default FW_PN_R_SLEW_MAX 50.0
+
```
Comment (mahima-yoga) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172:138]: No yaw tuning?

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+
+################################################################################
+# SECTION 11: ADVANCED FEATURES & SYSTEM CONFIGURATION
+################################################################################
+
+# Manual Control
+param set-default MPC_MAN_Y_MAX 20.0
+param set-default MPC_JERK_AUTO 1.0
+param set-default MPC_JERK_MAX 3.0
```
Comment (mahima-yoga) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172]: MPC is "Multicopter Position Control" :smile: , so these can be removed

**Review by mahima-yoga — COMMENTED** [review]
Thanks for such well structured airframe files :smile:. I didn't have time yet to go through everything

**Comment by alireza787b** [comment]
@mahima-yoga Good catches! Fixed both issues:

  1. ✅ Removed MPC_* parameters from Cessna 172 (fixed-wing shouldn't have multicopter position control params)

  2. ✅ Added yaw tuning to all fixed-wing/VTOL cruise modes:
     - Cessna 172: FW_YR_P=0.08, FW_YR_I=0.12 (large GA tuning)
     - TB2: FW_YR_P=0.06, FW_YR_I=0.15 (V-tail + endurance)
     - Alia-250: FW_YR_P=0.1, FW_YR_I=0.12, FW_Y_RMAX=60 (high-speed cruise)

  Note: Only parameters differing from PX4 defaults are included in configs (keeping files clean per your feedback on structure). Default 
  values (FW_YR_FF=0.3, FW_YR_D=0.0, FW_Y_RMAX=50 for Cessna/TB2) are omitted.
I will give each a further fine-tune while creating the demo video, and if I find better tuning values will commit again

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 (hunk 7 above)
Comment (alireza787b) [reply ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172]: done... thanks... missed that

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 (hunk 6 above)
Comment (alireza787b) [reply ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172:138]: Only parameters differing from PX4 defaults are included in configs (keeping files clean per your feedback on structure). Default
values (FW_YR_FF=0.3, FW_YR_D=0.0, FW_Y_RMAX=50 for Cessna/TB2) are omitted.
I will give each a further fine-tune while creating the demo video, and if I find better tuning values will commit again

Comment (mahima-yoga) [reply ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172:138]: > Only parameters differing from PX4 defaults are included in configs

That's true, just making sure it was intentional

File: ROMFS/px4fmu_common/init.d-posix/airframes/5021_xplane_qtailsitter [hunk 8]
```diff
@@ -0,0 +1,412 @@
...
+# Transition Safety
+param set-default VT_FW_MIN_ALT 30.0     # Minimum FW altitude (30m)
+param set-default VT_PITCH_MIN -8.0      # Min pitch angle in transition
+
+
+################################################################################
+# SECTION 13: FAILSAFE & SAFETY
+################################################################################
+
+param set-default FD_FAIL_R 70           # Failsafe return altitude
```
Comment (mahima-yoga) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5021_xplane_qtailsitter]: Maybe it's just the comment that's wrong but FD_FAIL_R is the roll angle at which attitude failure occurs

**Comment by hamishwillee** [comment]
@mahima-yoga Thanks so much for reviewing. When you're done let me know and I'll have another last look at the docs.

**Comment by mahima-yoga** [comment]
@StefanoColli could you still have a quick look from a simulation perspective? Otherwise to me it looks good!

**Comment by alireza787b** [comment]
I am going to do a final PID tune for each frame this week.  If there is anything else for PX4 standards, please let me know, and I will apply and make it final, ready to merge.

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 [hunk 9]
```diff
@@ -0,0 +1,338 @@
+#!/bin/sh
+################################################################################
+# PX4 SITL Configuration Script for Cessna 172 Skyhawk in X-Plane
+################################################################################
+#
+# @name              Cessna 172 Skyhawk
```
Comment (hamishwillee) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172]: The posix frames are not being generated in https://docs.px4.io/main/en/airframes/airframe_reference , which seems like an omission to me (asking in [discord here](https://discord.com/channels/1022170275984457759/1440866339479879752)). 

However they might be in future, so how about we add the name? (something similar for all)

```suggestion
# @name              Xplane Cessna 172 Skyhawk
```

File: docs/en/sim_xplane/index.md [hunk 10]
```diff
@@ -0,0 +1,529 @@
...
+- **Vehicle type agnostic**: Fixed-wing, multicopter, VTOL, helicopter, or hybrid configurations
+- **Custom parameter tuning**: Full access to PX4's parameter system for flight controller optimization
+
+See [Custom Aircraft](#custom-aircraft) section and the [custom airframe guide](https://github.com/alireza787b/px4xplane/blob/master/docs/custom-airframe-config.md) for detailed configuration instructions.
+
+## Supported Aircraft
+
+PX4 includes pre-configured airframes for X-Plane:
+
+| Airframe ID | Name               | Type        | Description                           |
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]: You might want to add the start command here in a column too (though they are listed below)

**Review by hamishwillee — DISMISSED** [review]
Docs look good to me. Approving, but let's wait on your final changes to merge

**Comment by hamishwillee** [comment]
You should check the [Checks / build (check_newlines) (pull_request)](https://github.com/PX4/PX4-Autopilot/actions/runs/19780376702/job/56679695360?pr=22493)
 fail and please let us know when you're ready for another look over

File: ROMFS/px4fmu_common/init.d-posix/airframes/5021_xplane_qtailsitter (hunk 8 above)
Comment (hamishwillee) [reply ROMFS/px4fmu_common/init.d-posix/airframes/5021_xplane_qtailsitter]: @alireza787b See this

**Comment by hamishwillee** [comment]
@mahima-yoga What else do you want from this?

**Comment by hamishwillee** [comment]
@mahima-yoga Can you confirm what needs to happen with this?

**Comment by alireza787b** [comment]
Thanks @hamishwillee, and sorry for the slow response. I had limited connectivity during the regional internet disruption, but I am back on this now.

From my side, the remaining work is to rebase/check against current main, resolve the stale CI failures, and confirm whether the standard_vtol SITL failure is caused by this PR or unrelated base behavior. After that I will ask for another docs/simulation review.

@mahima-yoga / @StefanoColli, if there are any remaining technical blockers for the airframe configs, please point me to them and I will address them.

**Comment by hamishwillee** [comment]
FYI @mahima-yoga is on holiday. Might take a little bit to respond.

**Review by mahima-yoga — DISMISSED** [review]
From an airframe perspective it looks good to me! 

Please either clean up the commits before merging, or squash merge

**Comment by hamishwillee** [comment]
Thanks for review @mahima-yoga ! @alireza787b Looks like we're good to go once the conflict fixed and commits are squashed

**Comment by hamishwillee** [comment]
@alireza787b As above "Looks like we're good to go once the conflict fixed and commits are squashed". Can you do that?

Also, is the tuning finished? @mahima-yoga reviewed this but if you change things too much we'll have to get another review.

**Comment by farhangnaderi** [comment]
@alireza787b Thanks. Whenever ready and conflicts are resolved as Hamish mentioned we can go for another review.

**Comment by alireza787b** [comment]
Thanks @hamishwillee and @farhangnaderi. Quick update from my side, because I know I have been quiet longer than expected.

I did not want to just squash the old branch and ask for another review before checking the whole stack again. Over the last round I went back through the X-Plane side and the PX4 side with new logs:

- rebased/synced against current PX4 main and fixed stale params / current SITL startup defaults
- re-tested and tuned the five airframes again: Alia, Ehang, QuadTailsitter, Cessna 172, and TB2
- checked startup/sensor timing, low-FPS / pause behavior, land detection, and the noisy startup logs found during testing
- updated the px4xplane-side config workflow, config validation/editor UX, camera config support, and test docs/cards outside this PX4 branch
- cleaned the PX4 airframe parameter files so the defaults are intentional and easier to review

I also cleaned the PR branch now: it is squashed to one commit, and the diff no longer carries the EKF2 core change or the temporary logger cleanup. This PR is now scoped to X-Plane SITL airframes, target wiring, the small simulator startup handling needed for X-Plane defaults, and docs.

During the final Alia transition tests against current main I found a separate EKF-GSF emergency yaw-reset edge case. In short: GNSS velocity aiding and magnetic heading were healthy, but a transient GNSS position rejection could still allow an EKF-GSF yaw reset just after VTOL handoff. In my logs that reset yaw to a bad GSF estimate and then produced a false compass fault.

I opened that as a focused EKF PR here:
https://github.com/PX4/PX4-Autopilot/pull/27533

That PR is passing CI now.

Question for reviewers: do you prefer that I continue finalizing this X-Plane PR independently while the EKF PR is reviewed separately, or should I treat this PR as temporarily waiting on the EKF PR for final VTOL transition validation?

File: docs/en/sim_xplane/index.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+| Ehang 184      | Multicopter     | `make px4_sitl_default xplane_ehang184`    |
+| Alia-250       | VTOL quadplane  | `make px4_sitl_default xplane_alia250`     |
+| QuadTailsitter | VTOL tailsitter | `make px4_sitl_default xplane_qtailsitter` |
+
+The matching X-Plane aircraft files and bridge configuration are distributed by the px4xplane project.
+
+## Installation
+
+1. Install the [PX4 development environment](../dev_setup/dev_env.md).
+2. Install X-Plane 11 or X-Plane 12.
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md:31]: IF there is a great installation link, would be good to provide.

File: docs/en/sim_xplane/index.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+3. Download a px4xplane release from the [px4xplane releases page](https://github.com/alireza787b/px4xplane/releases).
+4. Install the bridge plugin into:
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]: What bridge plugin are you refering to?

File: docs/en/sim_xplane/index.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+| Alia-250       | VTOL quadplane  | `make px4_sitl_default xplane_alia250`     |
+| QuadTailsitter | VTOL tailsitter | `make px4_sitl_default xplane_qtailsitter` |
+
+The matching X-Plane aircraft files and bridge configuration are distributed by the px4xplane project.
+
+## Installation
+
+1. Install the [PX4 development environment](../dev_setup/dev_env.md).
+2. Install X-Plane 11 or X-Plane 12.
+3. Download a px4xplane release from the [px4xplane releases page](https://github.com/alireza787b/px4xplane/releases).
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]: Which one? Basically I'm looking at these points and I think you're assuming that PX4 developers know something about X-Plane. Maybe they do, but a little hand holding and assuming they know nothing might be better.

File: docs/en/sim_xplane/index.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
+# X-Plane Simulation
+
+:::warning
+This simulator is [community supported and maintained](../simulation/community_supported_simulators.md).
+It may or may not work with current versions of PX4 and may be removed in future releases.
+See [Toolchain Installation](../dev_setup/dev_env.md) for information about the environments and tools supported by the core development team.
+:::
+
+[X-Plane](https://www.x-plane.com/) can be used with PX4 for [Software-In-The-Loop (SITL)](../simulation/index.md#sitl-simulation-environment) simulation through the community-supported [px4xplane](https://github.com/alireza787b/px4xplane) bridge plugin.
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md:9]: Does it work on any OS/all OS supported by PX4? Can I run it in WSL2?

File: docs/en/sim_xplane/index.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+1. Install the [PX4 development environment](../dev_setup/dev_env.md).
+2. Install X-Plane 11 or X-Plane 12.
+3. Download a px4xplane release from the [px4xplane releases page](https://github.com/alireza787b/px4xplane/releases).
+4. Install the bridge plugin into:
+
+   ```sh
+   X-Plane/Resources/plugins/px4xplane
+   ```
+
+5. Install the matching X-Plane aircraft and bridge configuration from the release package.
```
Comment (hamishwillee) [inline docs/en/sim_xplane/index.md]: Is this the package I just downloaded from " Download a px4xplane release " step? Is that a double click? The less users have to think the more likely they are to try it out.

**Comment by hamishwillee** [comment]
@mahima-yoga @farhangnaderi Can one of you please go through this quickly again as it has had lots of change since your last check, as outlined in https://github.com/PX4/PX4-Autopilot/pull/22493#issuecomment-4589317993.

**Comment by alireza787b** [comment]
Thanks @hamishwillee, good point. The page was assuming too much X-Plane/px4xplane context.

I pushed a small docs-only update in `23d6fa4`:

- clarified that the px4xplane bridge is the X-Plane plugin and what it does between PX4 and X-Plane
- made the release install flow explicit: download the OS-specific archive, extract it, copy the plugin folder, and copy the matching aircraft files
- clarified the Windows + WSL2 workflow vs same-machine Linux/macOS usage
- adjusted the community simulator row so it says these are the preconfigured PX4 targets, while custom aircraft need a matching PX4 airframe and px4xplane bridge config

I kept the PX4 docs intentionally short and linked out to the px4xplane docs for the plugin layout/custom-airframe details.

I also left your docs ordering commit in place. If you prefer the branch squashed again after this review pass, I can do that.

**Comment by alireza787b** [comment]
One more small docs-only follow-up in `271dc46` after rereading the setup page from a new user point of view. clarified that `PX4_SIM_HOSTNAME` is set on the PX4/SITL side and should point to the machine running X-Plane + the plugin, including WSL2/Docker/remote-host cases. I also added the TCP 4560 firewall note, linked the official X-Plane Plane Maker/dataref references, and adjusted the custom-model wording so it is clear the shipped targets are examples, while new models still need a matching PX4 control path and writable dataref mappings.\n\nNo code or parameter changes in this push.

**Comment by alireza787b** [comment]
Final update from my side before the next review pass: px4xplane v4.0.0 is now published with Windows, Linux, and macOS plugin packages: https://github.com/alireza787b/px4xplane/releases/tag/v4.0.0\n- the px4xplane README/docs were cleaned up for the current workflow, including WSL2/Docker host-IP setup, config editor docs, custom-airframe guidance, and the video archive/future playlist link\n- this PX4 PR is green in CI and still scoped to the X-Plane SITL targets/docs/startup wiring\n\nThe separate EKF-GSF yaw-reset issue found during final Alia VTOL transition testing is still isolated in #27533. That PR is also green in CI. Until that lands, a fast VTOL transition on current PX4 main can still hit the false compass/yaw-reset warning in that specific edge case, but I am keeping that fix separate from this X-Plane integration PR.

**Comment by hamishwillee** [comment]
Docs look better thanks.

> I also left your docs ordering commit in place. If you prefer the branch squashed again after this review pass, I can do that.

Fine by me. It will all be squashed on merge.

File: ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172 [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+################################################################################
+
+param set-default FW_WING_SPAN 11.0
+param set-default FW_WING_HEIGHT 3.0
+param set-default WEIGHT_BASE 1157.0
+param set-default WEIGHT_GROSS 1157.0
+
+param set-default MPC_MAN_Y_MAX 20.0
+param set-default MPC_JERK_AUTO 1.0
+param set-default MPC_JERK_MAX 3.0
```
Comment (mahima-yoga) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5001_xplane_cessna172]: These multicopter parameters can be removed.

File: ROMFS/px4fmu_common/init.d-posix/airframes/5002_xplane_tb2 [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+# Baykar lists 70 kt cruise class and 90-110 KTAS travel/max. The X-Plane model
+# is lighter than the published 700 kg MTOW, so these values are conservative
+# SITL targets.
+param set-default FW_AIRSPD_MIN 24.0
+param set-default FW_AIRSPD_MAX 56.0
+param set-default FW_AIRSPD_TRIM 36.0
+param set-default FW_AIRSPD_STALL 20.0
+param set-default FW_AIRSPD_FLP_SC 0.85
+param set-default FW_USE_AIRSPD 1
+param set-default ASPD_DO_CHECKS 1
```
Comment (mahima-yoga) [inline ROMFS/px4fmu_common/init.d-posix/airframes/5002_xplane_tb2:180]: How come all the other checks are disabled?

**Review by mahima-yoga — CHANGES_REQUESTED** [review]
There are a lot of parameters that are set to their default value - and this adds a lot of unnecessary bloat to the airframe files. Usually we only want to set the the non-default parameters. 

Some examples I found were: 

- Many EKF2 parameters 
- SYS_HAS parameters 
- FW_USE_AIRSPD 
- FW_THR_IDLE
- FW_THR_MAX
- RWTO_NUDGE
- LNDFW_TRIG_TIME

Would you mind removing the default parameters?

**Comment by alireza787b** [comment]
Thanks @mahima-yoga. I pushed `e1634873` for this.

I went through all five X-Plane airframe files and removed the params that were only restating inherited PX4 defaults. I checked this against the generated parameter defaults plus the relevant `rc.fw_defaults`, `rc.mc_defaults`, and `rc.vtol_defaults`, so the tuned values stay in the airframe files and the default duplicates are gone.

A couple of notes on intentional behavior:

- `ASPD_DO_CHECKS` is no longer overridden. Cessna, TB2, Alia, and QuadTailsitter now inherit the PX4 default checks instead of forcing only the missing-data check.
- `SYS_HAS_NUM_ASPD=0` is still kept for the X-Plane airspeed-enabled frames. The pitot data comes from the simulator/HIL stream, not a physical airspeed driver that should block prearm. The aircraft still use modeled airspeed for control/fallback where configured.
- The Cessna leftover `MPC_*` lines are removed. They were not part of the fixed-wing control path.

Local checks I ran before pushing:

- duplicate-default scan over all five X-Plane airframes: zero redundant inherited defaults remaining
- `git diff --check`
- `bash -n` on all five airframe scripts
- `make px4_sitl_default`
- launched `xplane_cessna172`, `xplane_tb2`, `xplane_ehang184`, `xplane_alia250`, and `xplane_qtailsitter` up to the simulator TCP wait; no parameter-not-found or shell errors

CI is running again now.

**Review by mahima-yoga — DISMISSED** [review]
Thanks for all the detailed descriptions, comments and tests!

**Comment by alireza787b** [comment]
> Thanks for all the detailed descriptions, comments and tests!

Thanks, I pushed one final validation update after retesting Alia and C172 against the current branch.

  This last push is still scoped to X-Plane airframe validation defaults: Alia front-transition pitch offset, and C172 climbout/flap/energy margins after runway takeoff. CI is
  running again now.

  During final VTOL validation I also isolated two PX4-side edge cases separately:
  - #27533 for the EKF-GSF yaw-reset/compass warning case seen in fast VTOL transition.
  - #27601 for Standard VTOL front-transition setpoint handoff. That one is still draft while I finish validation and get maintainer feedback.

  I am doing one last local end-to-end pass with those guard PRs applied locally before saying this is ready to merge. Please hold merge until I confirm the final validation
  result.

**Comment by alireza787b** [comment]
Final update from my side. The local end-to-end validation pass is done now, and I no longer need this PR held for more X-Plane-side changes.

What I rechecked after the last upstream sync:

- all five X-Plane targets: Alia 250, Cessna 172, TB2, Ehang 184, and QuadTailsitter
- clean SITL startup with fresh saved parameters / reset-config flow
- WSL2 host-IP workflow and the px4xplane launcher sync path
- X-Plane docs from a new-user point of view: bridge/plugin meaning, release install, WSL2/Docker/remote-host IP setup, TCP 4560, custom aircraft / Plane Maker / dataref mapping links
- airframe params after the duplicate-default cleanup requested by @mahima-yoga

This PR is still scoped to the X-Plane SITL integration: targets, airframes, startup wiring, and docs. While validating against current upstream PX4 I also found a few PX4-side controller/estimator edge cases, so I kept them out of this branch and split them into separate PRs:

- #27533 - EKF-GSF emergency yaw reset guard for a fast VTOL transition edge case
- #27601 - Standard VTOL front-transition pusher-thrust / pitch-setpoint handoff
- #27669 - Tailsitter fixed-wing attitude-frame consistency for quaternion attitude error
- #27670 - Fixed-wing TECS reset when the local altitude reference changes

Those separate PRs explain the delay in the final validation pass after rebasing/syncing to current PX4. I used them locally in the validation stack where needed, but they are intentionally not bundled into this X-Plane SITL PR.

From my side #22493 is ready for final review/merge. If there is any remaining blocker in this branch, please point me to it and I will fix it directly.

**Review by hamishwillee — DISMISSED** [review]
Thanks @alireza787b - @mahima-yoga Can you merge this if you are happy? The docs seem sufficient :-)

**Review by mahima-yoga — DISMISSED** [review]
So sorry that I let this one slip. Looks good to merge, if we find a small issue/mis-configuration we can always fix it later but I think it's time to bring this in now 😄 

Could you do a rebase to resolve the conflicts? And then let's merge it!

**Comment by alireza787b** [comment]
Thanks @mahima-yoga. I rebased this on current `main` and resolved the docs conflict by keeping both the upstream RotorPy entry and the X-Plane entry.\n\nLocal checks after the rebase:\n- `make px4_sitl_default`\n- `git diff --check`\n- conflict-marker scan over the touched docs/startup files\n\nThe branch is mergeable again now. CI restarted after the force-push.

**Review by hamishwillee — APPROVED** [review]
Thanks @alireza787b @mahima-yoga 

Merging based on @mahima-yoga review and request for rebase (but then merge).
